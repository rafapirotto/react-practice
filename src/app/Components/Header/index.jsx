import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';

import AuthenticatedHeader from './Components/AuthenticatedHeader';
import NonAuthenticatedHeader from './Components/NonAuthenticatedHeader';
import { isAuthenticated } from '../../utils';

const Header = () => {
  const { state } = useSelector((state) => state.login);

  useEffect(() => {}, [state]);

  return isAuthenticated() ? <AuthenticatedHeader /> : <NonAuthenticatedHeader />;
};

export default Header;
