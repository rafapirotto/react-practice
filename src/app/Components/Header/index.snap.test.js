import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Header from '.';

test('It matches the snapshot', () => {
  const component = renderer.create(
    <Router>
      <Header />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
