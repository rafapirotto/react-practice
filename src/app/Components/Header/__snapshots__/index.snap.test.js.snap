// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<header
  className="flex ml-24 pr-10 py-20 w-11/12 z-10 absolute"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="movy text-4xl"
    >
      Movy
    </div>
  </a>
  <div
    className="flex ml-32 items-center flex-wrap"
  >
    <div
      className="mx-6 font-normal"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="mx-6"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Movies
      </a>
    </div>
    <div
      className="mx-6"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Series
      </a>
    </div>
    <div
      className="mx-6"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Recently Added
      </a>
    </div>
    <div
      className="mx-6"
    >
      <a
        href="/my-list"
        onClick={[Function]}
      >
        My List
      </a>
    </div>
  </div>
  <div
    className="flex items-center flex-grow justify-end"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="search"
        className="mx-3 search"
        src="search.png"
      />
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="user"
        className="mx-3 userImage"
        src="user-image.png"
      />
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="arrow"
        className="mx-3 arrow"
        src="arrow.png"
      />
    </a>
  </div>
</header>
`;
